source/common/tcp_proxy/tcp_proxy.cc

```diff
@@ -71,6 +77,17 @@ Config::Config(const envoy::config::filter::network::tcp_proxy::v2::TcpProxy& co
     routes_.emplace_back(default_route);
   }
 
+  // Weighted clusters will be enabled only if both the default cluster and
+  // deprecated v1 routes are absent.
```

Comment:
Ok. What should be the routing order between weighted and v1 routes?

Reply:
I think the only way would be to use weighted clusters as a fallback for v1 routes then.

Reply:
No I meant we get rid of the code for v1 routes completely. I was the one who added it because Istio needed it since filter chain was not fully implemented back then. Once that landed, we switched the internal implementation in Pilot to use filter chain selection for TCP proxy and got rid of v1 routes.

Reply:
Got it, will do.

Reply:
Let's remove routes as a separate PR.

Reply:
But if you remove routes first, this PR gets slightly simpler. But I'm fine with either order for the PRs.

Reply:
Created #4446 for removing v1 routes support.